views/building-hours/building.js

```diff
@@ -6,86 +6,101 @@
 
 import React from 'react'
 import {
-  StyleSheet,
-  View,
-  Text,
-  Image,
+ View,
+ Text,
+ Image,
+ Dimensions,
+ StyleSheet,
 } from 'react-native'
 
 import * as c from '../components/colors'
 import type {BuildingInfoType} from './types'
 import {isBuildingOpen} from './is-building-open'
 import {formatBuildingHours} from './format-building-hours'
+import CollapsibleBlock from '../components/collapsibleBlock'
+import {allBuildingHours} from './all-building-hours.js'
 
-let styles = StyleSheet.create({
-  container: {
-    flex: 1,
-    justifyContent: 'center',
-    borderLeftWidth: 5,
-    borderRightWidth: 5,
-    margin: 5,
-    height: 100,
-  },
-  buildingOpen: {
-    borderColor: c.pastelGreen,
-  },
-  buildingClosed: {
-    borderColor: c.strawberry,
-  },
-  buildingAlmostClosed: {
-    borderColor: c.mustard,
-  },
-  name: {
-    color: c.white,
-    fontSize: 24,
-    textAlign: 'center',
-    backgroundColor: 'rgba(0, 0, 0, 0.65)',
-  },
-  inner: {
-    flex: 1,
-    flexDirection: 'column',
-    justifyContent: 'space-between',
-  },
-  status: {
-    color: c.white,
-    fontSize: 14,
-    paddingTop: 2,
-    paddingBottom: 2,
-    textAlign: 'center',
-    backgroundColor: 'rgba(0, 0, 0, 0.45)',
-  },
-})
 
-let borderColors = {
-  'Open': styles.buildingOpen,
-  'Almost Closed': styles.buildingAlmostClosed,
-  'Closed': styles.buildingClosed,
-}
-
-type PropsType = {
-  info: BuildingInfoType,
-  image: number,
-  name: string,
-};
 export default function BuildingView({info, image, name}: PropsType) {
-  let open = isBuildingOpen(info)
-  let borderColor = borderColors[open]
+
+  let borderColors = {
+    'Open': '#CEFFCE',
+    'Almost Closed': '#FFFC96',
+    'Closed': '#F7C8C8',
+  }
+
+  type PropsType = {
+    info: BuildingInfoType,
+    image: number,
+    name: string,
+  };
+
+  let openStatus = isBuildingOpen(info)
+
+  let styles = StyleSheet.create({
+    container: {
+      flex: 1,
+      justifyContent: 'center',
+      borderLeftWidth: 5,
+      borderRightWidth: 5,
+      marginTop: 10,
+      height: 100,
+      borderColor: borderColors[openStatus],
+    },
+    name: {
+      color: c.white,
+      fontSize: 24,
+      textAlign: 'center',
+      backgroundColor: 'rgba(0, 0, 0, 0.65)',
+    },
+    inner: {
+      flex: 1,
+      flexDirection: 'column',
+      justifyContent: 'space-between',
+    },
+    status: {
+      color: c.white,
+      fontSize: 14,
+      paddingTop: 2,
+      paddingBottom: 2,
+      textAlign: 'center',
+      backgroundColor: 'rgba(0, 0, 0, 0.45)',
+    },
+    hoursText: {
+      textAlign: 'center',
+      fontSize: 13,
+    },
+    contents: {
+      paddingBottom: 10,
+      paddingTop: 5,
+      width: Dimensions.get('window').width,
```

Comment:
Oh, and excluding this, too, since it needs to be able to respond to size changes.